modules/imgproc/include/opencv2/imgproc/imgproc_c.h

```diff
@@ -435,12 +441,13 @@ CVAPI(int)  cvFindContours( CvArr* image, CvMemStorage* storage, CvSeq** first_c
                             int method CV_DEFAULT(CV_CHAIN_APPROX_SIMPLE),
                             CvPoint offset CV_DEFAULT(cvPoint(0,0)));
 
-/** @brief Initializes contour retrieving process.
+CVAPI(int)  cvFindContours2( CvArr* image, CvMemStorage* storage, CvSeq** first_contour,
+                            int header_size CV_DEFAULT(sizeof(CvContour)),
+                            int mode CV_DEFAULT(CV_RETR_LIST),
+                            int method CV_DEFAULT(CV_CHAIN_APPROX_SIMPLE),
+                            CvPoint offset CV_DEFAULT(cvPoint(0,0)), int needFillBorder CV_DEFAULT(1));
```

Comment:
My suggestion is to make these functions internal only (use 'static' in .cpp file).
BTW, You may use "Impl" suffix instead of "2".